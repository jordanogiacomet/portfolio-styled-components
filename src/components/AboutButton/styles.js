import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
