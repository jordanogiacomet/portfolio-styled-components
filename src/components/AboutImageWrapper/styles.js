import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
