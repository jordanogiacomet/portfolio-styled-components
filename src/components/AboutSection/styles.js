import styled from 'styled-components';

export const Container = styled.section`
  padding: 50px 0;
  display: flex;
  flex-direction: column;

  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }

  @media (min-width: 1280px) {
    gap: 50px;
  }
`;

