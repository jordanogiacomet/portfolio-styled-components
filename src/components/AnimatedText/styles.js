import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: black;
  }
  &.dark{
    color: white;
  }
`;