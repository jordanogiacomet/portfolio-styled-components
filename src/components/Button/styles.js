import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 