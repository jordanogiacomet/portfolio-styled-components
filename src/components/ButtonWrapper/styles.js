import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;