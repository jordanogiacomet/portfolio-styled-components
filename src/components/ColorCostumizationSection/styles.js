// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
