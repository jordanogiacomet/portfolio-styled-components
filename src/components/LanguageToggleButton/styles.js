import styled from 'styled-components';

export const Container = styled.button`
  padding: 1rem;
  background-color: transparent;
  box-shadow: none;
  border: 1px solid gray;
  border-radius: 9999px;
  color: white;
  font-size: 1.2rem;
  position: absolute;
  right: 2.5rem;
  cursor: pointer;
  transition: 0.2s;
  &:hover{
    color: #4169E1;
    border-color: #4169E1;
  }
`;
