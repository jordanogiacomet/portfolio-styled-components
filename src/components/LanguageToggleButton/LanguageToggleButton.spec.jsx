import { screen, fireEvent } from '@testing-library/react';
import { LanguageToggleButton } from '.';
import { renderTheme } from '../../styles/render-theme';
import { LanguageProvider } from '../../contexts/LanguageContext';
import { ColorModeProvider } from '../../contexts/ColorModeContext';

describe('<LanguageToggleButton />', () => {
  it('should render with initial language', () => {
    renderTheme(
      <ColorModeProvider>
        <LanguageProvider>
          <LanguageToggleButton />
        </LanguageProvider>
      </ColorModeProvider>
    );
    expect(screen.getByText('PT-BR')).toBeInTheDocument();
  });

  it('should toggle language on click', () => {
    renderTheme(
      <ColorModeProvider>
        <LanguageProvider>
          <LanguageToggleButton />
        </LanguageProvider>
      </ColorModeProvider>
    );
    const button = screen.getByText('PT-BR');
    fireEvent.click(button);
    expect(screen.getByText('EN-US')).toBeInTheDocument();
  });

  it('should have the correct aria-label', () => {
    renderTheme(
      <ColorModeProvider>
        <LanguageProvider>
          <LanguageToggleButton />
        </LanguageProvider>
      </ColorModeProvider>
    );
    expect(screen.getByLabelText('Switch language')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <ColorModeProvider>
        <LanguageProvider>
          <LanguageToggleButton />
        </LanguageProvider>
      </ColorModeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
