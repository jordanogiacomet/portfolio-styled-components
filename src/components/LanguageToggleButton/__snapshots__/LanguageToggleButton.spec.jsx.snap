// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageToggleButton /> should match snapshot 1`] = `
.c0 {
  padding: 1rem;
  background-color: transparent;
  box-shadow: none;
  border: 2px solid #ADB7BE;
  border-radius: 9999px;
  color: #ADB7BE;
  font-size: 1.2rem;
  right: 2.5rem;
  margin-right: 2rem;
  cursor: pointer;
  transition: 0.2s;
}

.c0.light {
  color: white;
  border-color: white;
}

.c0.light:hover {
  color: #CCCCCC;
  border-color: #CCCCCC;
}

.c0:hover {
  color: #4169E1;
  border-color: #4169E1;
}

@media (max-width: 768px) {
  .c0 {
    bottom: 20px;
  }
}

<div>
  <button
    aria-label="Switch language"
    class="c0 dark"
  >
    PT-BR
  </button>
</div>
`;
